src/bucket/BucketIndexImpl.cpp

```diff
@@ -100,6 +101,79 @@ BucketIndexImpl<IndexT>::BucketIndexImpl(BucketManager& bm,
         std::vector<uint64_t> keyHashes;
         auto seed = shortHash::getShortHashInitKey();
 
+        auto countEntry = [&](BucketEntry const& be) {
+            auto ledgerEntryToLedgerEntryAndDurabilityType =
```

Comment:
Can we move this inner lambda to be a static function instead? Generally speaking I'd like to avoid nested lambdas for clarity. It might make the most sense to move this static function and `LedgerEntryTypeAndDurability` definition to `LedgerTypeUtils`. Also, I don't think we need to pass in `isTemporaryEntry` at all here. The simplest interface would be something like `static LedgerEntryTypeAndDurability getLedgerEntryAndDurabilityType(BucketEntry const& be)`.